src/item/pickit.py

```diff
@@ -77,10 +78,13 @@ def pick_up_items(self, char: IChar, is_at_trav: bool = False) -> bool:
                     time.sleep(0.2)
             else:
                 found_nothing = 0
-                closest_item = item_list[0]
-                for item in item_list[1:]:
-                    if closest_item.dist > item.dist:
-                        closest_item = item
+                item_list.sort(key=itemgetter('dist'))
+                for pattern in ["rune_", "uniq_", "set_"]:
+                    closest_item = next((obj for obj in item_list if pattern in obj["name"]), None)
+                    if closest_item:
+                        break
+                if not closest_item:
+                    closest_item = item_list[0]
```

Comment:
Doesn't seem like were doing to closest check on things that don't match the patterns anymore? Not sure if that matters or not as probably less important after the pattern items are grabbed.

Otherwise looks good to me

Reply:
Yeah since the list is sorted by distance before the loop it'll be fine